### PR TITLE
Various asf.yaml updates

### PR DESCRIPTION
**Pull request by janhoy**
I browsed the [asf.yaml docs](https://cwiki.apache.org/confluence/pages/viewpage.action?spaceKey=INFRA&title=git+-+.asf.yaml+features), and found several new capabilities related to GitHub that I believe we should add:

* `protected_branches` - our main branch should at least be protected from force-push
* `protected_tags` - prevent deletion of `release*` tags
* `autolink_jira` - make references to short-form SOLR JIRA issues in PRs clickable
* `collaborators` - list of github user names to allow full PR edit karma

### CODE REVIEW

File: .asf.yaml
Comment (cpoerschke) [inline .asf.yaml:31]: > * `collaborators` - list of github user names to allow full PR edit karma

Am curious if that would cover its own PRs only or others' pull requests too, and/or when it (solrbot) would have a need to use full PR edit.

Comment (janhoy) [reply .asf.yaml:31]: The list of users here are allowed full edit rights for all issue/PRs, so it will be users who are not yet committers, but we trust them to do triage. Solrbot would need this right in order to set a label on the PRs it opens. In [`renovate-config.js`](https://github.com/solrbot/renovate-github-action/blob/main/solr/renovate-config.js) we could [configure it to set a `dependencies` label ](https://docs.renovatebot.com/configuration-options/#addlabels) and even labels detailing what kind of dependency it is (e.g. test vs compile). But that is disabled as of now as solrbot does not have that karma.

We could imagine future bots to assign reviewers to PRs intelligently, and such a bot would also be added here. But primarily it would be cool to get 3-5 non-committer developers to volunteer to help with PRs.

**Comment by janhoy** [comment]
Also, I consider `required_conversation_resolution`, which will require all conversations to be resolved before merging is possible. I have seen too quick merges of looong PRs where the author missed an open conversation.

File: .asf.yaml
Comment (dsmiley) [inline .asf.yaml:22]: Is there a way to add `branch_*`?

**Review by dsmiley — APPROVED** [review]

File: .asf.yaml
Comment (janhoy) [reply .asf.yaml:22]: I cannot see it documented. But we could easily enumerate all of them and let the release wizard add new branches to the list?

Comment (janhoy) [reply .asf.yaml:23]: Also, `.asf.yaml` does not support glob format (see https://the-asf.slack.com/archives/CBX4TSBQ8/p1668009454564229?thread_ts=1667901002.937809&cid=CBX4TSBQ8), but will probably be added once GH api supports it.

**Pull request merged**
